html5/kubeinvaders.js

```diff
@@ -199,77 +217,75 @@ function currentChaosContainerJsonTextAreaVal() {
-            codename = this.responseText.trim();
-            $("#" + elementId).val(codename);
-            $("#" + elementId).text(codename);
-            if (codename == "") {
-                $('#alert_placeholder').replaceWith(alert_div + 'Error getting codename from backend. </div>');
-                codename = "error_fix_getcodename_from_backend";
-            }
-        }
-    };;
-    oReq.open("GET", k8s_url + "/codename");
-    oReq.send();
+    const value = (await response.text()).trim();
+
+    $("#" + elementId).val(value);
+    $("#" + elementId).text(value);
+
+    if (value == "") {
+        showErrorAlert('Error getting codename from backend. ', "error_fix_getcodename_from_backend");
+    } else {
+        codename = value;
+    }
 }
 
-function getMetrics() {
-    var oReq = new XMLHttpRequest();
-    oReq.onload = function () {
-        var lines = this.responseText.split('\n');
-        for (var i = 0;i < lines.length;i++){
-            metric = lines[i].split(' ');
+async function getMetrics() {
+    const response = await httpGetToK8S("/metrics");
 
-            if (metric[0] == "chaos_node_jobs_total") {
-                $('#chaos_jobs_total').text(metric[1]);
-                chart_chaos_jobs_total = Number(metric[1]);
-            }
-            else if (metric[0] == "deleted_pods_total") {
-                chart_deleted_pods_total = Number(metric[1]);
-                $('#deleted_pods_total').text(metric[1]);            
-            }
-            else if (metric[0] == "fewer_replicas_seconds") {
-                chart_fewer_replicas_seconds = Number(metric[1]);
-                $('#fewer_replicas_seconds').text(metric[1]);            
-            }
-            else if (metric[0] == "latest_fewer_replicas_seconds") {
-                chart_latest_fewer_replicas_seconds = Number(metric[1]);
-                $('#latest_fewer_replicas_seconds').text(metric[1]);            
-            }
-            else if (metric[0] == "pods_not_running_on_selected_ns") {
-                chart_pods_not_running_on = Number(metric[1]);
-                $('#pods_not_running_on').text(metric[1]);            
-            }
-            else if (metric[0] == "pods_match_regex:" + random_code) {
-                $('#pods_match_regex').text(metric[1]);            
-            }
-            else if (metric[0].match(chaos_job_regex)) {
-                metrics_split = metric[0].split(":");
-                chaos_jobs_status.set(metrics_split[1] + ":" + metrics_split[2] + ":" +  metrics_split[3], metric[1]);
-            }
-            else if (metric[0] == "current_chaos_job_pod") {
-                chart_current_chaos_job_pod = Number(metric[1]);
-                $('#current_chaos_job_pod').text(metric[1]);
-            }
+    if (!response.ok) {
+        return;
+    }
+
+    const text = await response.text();
+
+    for (const line of text.split('\n')) {
+        const [key, value] = line.split(' ');
```

Comment:
Switched from:
```
        var lines = this.responseText.split('\n');
        for (var i = 0;i < lines.length;i++){
            metric = lines[i].split(' ');
```

Also I'm using the [deconstructing assignement](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Destructuring_assignment) to give a name to the key/value pair